values/encoding/int_encode.go

```diff
@@ -42,39 +41,45 @@ type intEncoder struct {
 // NewIntEncoder creates a new int encoder.
 func NewIntEncoder() IntEncoder {
 	return &intEncoder{
-		// Make buf at least big enough to hold Uint64 values.
-		buf: make([]byte, uint64SizeBytes),
+		// Make buf at least big enough to hold varint64 values.
+		buf: make([]byte, binary.MaxVarintLen64),
```

Comment:
No need to store `buf` as a field, you can define `buf` on the stack in `rawEncode`:
```
var b [binary.MaxVarintLen64]byte
buf := b[:]
```
then just use `buf` where `enc.buf` is used.

Reply:
The `buf` is used elsewhere as well. (e.g. for encoding the proto metadata).